kyuubi-common/src/test/scala/org/apache/kyuubi/session/SessionManagerSuite.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.session
+
+import java.time.Duration
+import java.util
+
+import org.apache.hive.service.rpc.thrift._
+import org.apache.thrift.protocol.TBinaryProtocol
+import org.apache.thrift.transport.TSocket
+
+import org.apache.kyuubi.{KyuubiFunSuite, Utils}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.service.NoopServer
+import org.apache.kyuubi.service.authentication.PlainSASLHelper
+
+class SessionManagerSuite extends KyuubiFunSuite {
+
+  private val server = new NoopServer()
+  private val conf = KyuubiConf()
+    .set(KyuubiConf.FRONTEND_BIND_PORT, 0)
+    .set("kyuubi.test.server.should.fail", "false")
+
+  val user: String = System.getProperty("user.name")
+
+  def startServer(): Unit = {
+    server.initialize(conf)
+    server.start()
+  }
+
+  override def afterAll(): Unit = {
+    server.getServices.foreach(_.stop())
+    super.afterAll()
+  }
+
+  protected def withThriftClient(f: TCLIService.Iface => Unit): Unit = {
```

Comment:
can we reuse the existing code instead of copying them everywhere？

Reply:
OK，I will solve this by extending 'FrontendServiceSuite'.